raft/src/main/java/org/apache/kafka/raft/RaftConfig.java

```diff
@@ -37,49 +36,56 @@
-    private static final String QUORUM_ELECTION_TIMEOUT_MS_DOC = "Maximum time in milliseconds to wait " +
+    public static final String QUORUM_ELECTION_TIMEOUT_MS_DOC = "Maximum time in milliseconds to wait " +
         "without being able to fetch from the leader before triggering a new election";
+    public static final int DEFAULT_QUORUM_ELECTION_TIMEOUT_MS = 5_000;
 
     public static final String QUORUM_FETCH_TIMEOUT_MS_CONFIG = QUORUM_PREFIX + "fetch.timeout.ms";
-    private static final String QUORUM_FETCH_TIMEOUT_MS_DOC = "Maximum time without a successful fetch from " +
+    public static final String QUORUM_FETCH_TIMEOUT_MS_DOC = "Maximum time without a successful fetch from " +
         "the current leader before becoming a candidate and triggering a election for voters; Maximum time without " +
         "receiving fetch from a majority of the quorum before asking around to see if there's a new epoch for leader";
+    public static final int DEFAULT_QUORUM_FETCH_TIMEOUT_MS = 15_000;
 
     public static final String QUORUM_ELECTION_BACKOFF_MAX_MS_CONFIG = QUORUM_PREFIX + "election.backoff.max.ms";
-    private static final String QUORUM_ELECTION_BACKOFF_MAX_MS_DOC = "Maximum time in milliseconds before starting new elections. " +
+    public static final String QUORUM_ELECTION_BACKOFF_MAX_MS_DOC = "Maximum time in milliseconds before starting new elections. " +
         "This is used in the binary exponential backoff mechanism that helps prevent gridlocked elections";
+    public static final int DEFAULT_QUORUM_ELECTION_BACKOFF_MAX_MS = 5_000;
 
     public static final String QUORUM_LINGER_MS_CONFIG = QUORUM_PREFIX + "append.linger.ms";
-    private static final String QUORUM_LINGER_MS_DOC = "The duration in milliseconds that the leader will " +
+    public static final String QUORUM_LINGER_MS_DOC = "The duration in milliseconds that the leader will " +
         "wait for writes to accumulate before flushing them to disk.";
+    public static final int DEFAULT_QUORUM_LINGER_MS = 25;
 
-    // Package-private for testing
-    static final String QUORUM_REQUEST_TIMEOUT_MS_CONFIG = QUORUM_PREFIX +
+    public static final String QUORUM_REQUEST_TIMEOUT_MS_CONFIG = QUORUM_PREFIX +
         CommonClientConfigs.REQUEST_TIMEOUT_MS_CONFIG;
+    public static final String QUORUM_REQUEST_TIMEOUT_MS_DOC = CommonClientConfigs.REQUEST_TIMEOUT_MS_DOC;
+    public static final int DEFAULT_QUORUM_REQUEST_TIMEOUT_MS = 20_000;
 
-    // Package-private for testing
-    static final String QUORUM_RETRY_BACKOFF_MS_CONFIG = QUORUM_PREFIX +
+    public static final String QUORUM_RETRY_BACKOFF_MS_CONFIG = QUORUM_PREFIX +
         CommonClientConfigs.RETRY_BACKOFF_MS_CONFIG;
+    public static final String QUORUM_RETRY_BACKOFF_MS_DOC = CommonClientConfigs.RETRY_BACKOFF_MS_DOC;
+    public static final int DEFAULT_QUORUM_RETRY_BACKOFF_MS = 100;
 
     static {
         CONFIG = new ConfigDef()
```

Comment:
Do we need to keep the definitions here?

Reply:
Raft cannot depend on core. So, it needs an implementation of AbstractConfig. ConfigDef here is necessary for that implementation. Is that what you meant by definitions?

Reply:
I guess what I'm saying is that RaftConfig does not need to extend AbstractConfig if the definitions are in KafkaConfig. Instead, we can let the RaftConfig constructor take an AbstractConfig, which will be the reference to KafkaConfig. Does that make sense?

Reply:
Right. That would be fine except that would mean the tests in the raft module would then have to depend on the core module to create a KafkaConfig out of the props. There's no circular dependency (between the raft test module and core), but that would mean we would introduce some sort of dependency on core in raft.

Reply:
`RaftConfig` just becomes a convenient holder for config values. We can have two constructors:
```java
public RaftConfig(AbstractConfig config)
public RaftConfig(String quorumVoters, int requestTimeoutMs, ...)
```

Reply:
Yea, that makes sense. I think devolving it to a util class of sorts (parsing the voter nodes/ids/connections and such) would dedupe the ConfigDef. I'll go ahead and make this change.

Reply:
Fixed. @hachikuji